text/0025-core-compute-resources-as-plugins.md

```diff
@@ -0,0 +1,77 @@
+- Feature Name: Core Compute Resources as Plugins
+- Start Date: 2016-10-21
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Core Compute Resources should move to plugins
+
+# Motivation
+[motivation]: #motivation
+
+Core has been set up to allow compute resources as plugins for some while, and
+we already have several examples of this in use (Xen, Azure, DigitalOcean,
+etc). This has brought benefits:
+
+* able to release independatly of Core
+* make use of the modular Fog gems that are now available
+* separate logging of issues
+* dedicated maintainers who know the compute resource well
+
+In addition, it allows the core project to mark a particular resource as
+unmaintained when necessary (as happened with Xen). 
+
+By comparison, some of the core compute resources are not well maintained, but
+users have an expectation that they work, because we ship them. By moving them
+to plugins, we gain the ability to properly support the maintainers or mark
+them as curerntly in need of a maintainer. It's also easier for new
+contributors to work on a specific resource they know, without having to learn
+the whole core codebase.
```

Comment:
Yes, but you have much less to work with (and potentially break - see https://github.com/theforeman/rfcs/pull/28#discussion_r85504538 about big vs small codebases). It's not about actual complexity.

Also, if the CR Plugin API is improved (as per https://github.com/theforeman/rfcs/pull/28#discussion_r85363994) and well documented, then it should be much much easier to work with.